### PR TITLE
improve getToDistributedCoinsFromGauges

### PR DESCRIPTION
**Pull request by catShaark**
Closes: #2005

## What is the purpose of the change

Change `getToDistributeCoinsFromGauges` so that it only iterate through all gauges once.

## Testing and Verifying

This change is already covered by existing tests

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? no
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? no
  - How is the feature or change documented?  not documented

### CODE REVIEW

File: x/incentives/keeper/distribute.go [hunk 1]
```diff
@@ -21,9 +21,14 @@ func (k Keeper) getDistributedCoinsFromGauges(gauges []types.Gauge) sdk.Coins {
 }
 
 func (k Keeper) getToDistributeCoinsFromGauges(gauges []types.Gauge) sdk.Coins {
-	// TODO: Consider optimizing this in the future to only require one iteration over all gauges.
-	coins := k.getCoinsFromGauges(gauges)
```
Comment (stackman27) [inline x/incentives/keeper/distribute.go:25]: `func (k Keeper) getCoinsFromGauges(gauges []types.Gauge) sdk.Coins {..}` is unused now, maybe you can remove it

**Review by stackman27 — APPROVED** [review]
LGTM once we figure out what to do with `getCoinsFromGauges()` function and linter fix😅

File: x/incentives/keeper/distribute.go (hunk 1 above)
Comment (catShaark) [reply x/incentives/keeper/distribute.go:25]: thanks, I'll remove it

**Review by alexanderbez — APPROVED** [review]

**Review by mattverse — APPROVED** [review]
Nice improvement with the complexity! LGTM!

**Pull request merged**